dissect/target/plugins/os/windows/tasks.py

```diff
@@ -58,452 +61,124 @@
-                    header_name=email_headers_name,
-                    header_value=email_headers_value,
-                    body=email_body,
-                    attachment=email_attachments,
-                )
-
-            if action_type == "ShowMessage":
-                title = self.get_element("Actions/ShowMessage/Title")
-                body = self.get_element("Actions/ShowMessage/Body")
-                yield ShowMessageRecord(
-                    action_type=action,
-                    title=title,
-                    body=body,
-                )
-
 
 class TasksPlugin(Plugin):
-    """TODO."""
+    """Plugin for retrieving scheduled tasks on a Windows system.
+
+    Args:
+        target: The target system.
+    """
+
+    ATTRIBUTES = [
+        "uri",
+        "security_descriptor",
+        "source",
+        "date",
+        "last_run_date",
+        "author",
+        "version",
+        "description",
+        "documentation",
+        "principal_id",
+        "user_id",
+        "logon_type",
+        "display_name",
+        "run_level",
+        "process_token_sid_type",
+        "required_privileges",
+        "allow_start_on_demand",
+        "restart_on_failure_interval",
+        "restart_on_failure_count",
+        "mutiple_instances_policy",
+        "dissalow_start_on_batteries",
+        "stop_going_on_batteries",
+        "start_when_available",
+        "network_profile_name",
+        "run_only_network_available",
+        "wake_to_run",
+        "enabled",
+        "hidden",
+        "delete_expired_task_after",
+        "idle_duration",
+        "idle_wait_timeout",
+        "idle_stop_on_idle_end",
+        "idle_restart_on_idle",
+        "network_settings_name",
+        "network_settings_id",
+        "execution_time_limit",
+        "priority",
+        "run_only_idle",
+        "unified_scheduling_engine",
+        "disallow_start_on_remote_app_session",
+        "data",
+        "_target",
+    ]
 
     PATHS = [
         "sysvol/windows/system32/tasks",
         "sysvol/windows/system32/tasks_migrated",
         "sysvol/windows/syswow64/tasks",
+        "sysvol/windows/tasks",  # at.exe job file location
     ]
 
-    def __init__(self, target):
+    def __init__(self, target: Target):
         super().__init__(target)
-        self.files = []
+        self.task_objects = []
 
-        for path in self.PATHS:
-            fpath = self.target.fs.path(path)
+        for file_path in self.PATHS:
+            fpath = self.target.fs.path(file_path)
             if not fpath.exists():
                 continue
 
             for entry in fpath.rglob("*"):
-                if entry.is_file():
-                    self.files.append(entry)
+                if entry.is_file() and entry.suffix.lower() == ".job":
+                    self.task_objects.append(AtTask(entry, target))
+                elif entry.is_file() and not entry.suffix:
+                    self.task_objects.append(XmlTask(entry, target))
```

Comment:
I wouldn't do the parsing of the tasks files on initialization but defer it until it is really needed.